sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/LogicalRelation.scala

```diff
@@ -72,9 +72,12 @@ case class LogicalRelation(
   // expId can be different but the relation is still the same.
   override lazy val cleanArgs: Seq[Any] = Seq(relation)
 
-  @transient override lazy val statistics: Statistics = Statistics(
-    sizeInBytes = BigInt(relation.sizeInBytes)
-  )
+  // statistics inherited from a MetastoreRelation
+  @transient var inheritedStats: Option[Statistics] = None
```

Comment:
This is not a bug in your PR. Let me fix it in a separate PR and then you can focus on the major goal of this PR.